### PR TITLE
refac: Incoming messages module

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR will introduce the IncomingMessages module, which handles:
- sync validation
- TransactionId and MessageId is unique
- Create a ServicebusMessage for the Process module to receive.

### CODE REVIEW

File: source/IncomingMessages.Application/Configuration/IncomingMessagesConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        services.AddScoped<IMessageParser, XmlMessageParser>();
+        services.AddScoped<IMessageParser, JsonMessageParser>();
+        services.AddScoped<IMessageParser, ProtoMessageParser>();
+        services.AddScoped<RequestAggregatedMeasureDataMarketMessageParser>();
+        services.AddTransient<SenderAuthorizer>();
+        services.AddTransient<IncomingRequestAggregatedMeasuredDataSender>();
+        services.AddTransient<RequestAggregatedMeasureDataValidator>();
+        services.AddScoped<ProcessTypeValidator>();
+        services.AddScoped<MessageTypeValidator>();
+        services.AddScoped<BusinessTypeValidator>();
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Application/Configuration/IncomingMessagesConfiguration.cs:44]: Can the validators be singleton?

File: source/IncomingMessages.Infrastructure/RequestAggregatedMeasureDataParsers/ProtoMessageParser.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public DocumentFormat HandledFormat => DocumentFormat.Proto;
+
+    public Task<RequestAggregatedMeasureDataMarketMessageParserResult> ParseAsync(
+        Stream message,
+        CancellationToken cancellationToken)
+    {
+        var requestAggregatedMeasureData = RequestAggregatedMeasureData.Parser.ParseFrom(message);
+        var marketMessage = RequestAggregatedMeasureDataMarketMessageFactory.Create(
+            requestAggregatedMeasureData,
+            SystemClock.Instance.GetCurrentInstant());
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Infrastructure/RequestAggregatedMeasureDataParsers/ProtoMessageParser.cs:41]: Can we make use of "ISystemDateTimeProvider" instead?

File: source/IntegrationTests/Infrastructure.CimMessageAdapter/Messages/RequestAggregatedMeasureData/RequestAggregatedMeasureDataReceiverTests.cs [hunk 3]
```diff
@@ -456,17 +450,11 @@ public async Task Multiple_activity_records_are_committed_as_processes()
...
         var processes = _processContext.AggregatedMeasureDataProcesses.ToList();
-        Assert.True(result.Success);
         Assert.NotNull(processes);
-        Assert.Equal(messageParserResult.MarketMessage!.Series.Count, processes.Count);
-
-        var document = messageParserResult.MarketMessage!;
-        await AssertTransactionIdIsStored(messageParserResult.MarketMessage!.SenderNumber, document.Series.First().Id);
-        await AssertTransactionIdIsStored(messageParserResult.MarketMessage!.SenderNumber, document.Series.Last().Id);
-        await AssertMessageIdIsStored(messageParserResult.MarketMessage!.SenderNumber, messageParserResult.MarketMessage!.MessageId);
+        Assert.Equal(2, messageParserResult.MarketMessage!.Series.Count);
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Infrastructure.CimMessageAdapter/Messages/RequestAggregatedMeasureData/RequestAggregatedMeasureDataReceiverTests.cs:457]: We want to make sure that we have 2 processes, not that we have two series when we are done parsing the request?

File: source/IncomingMessages.Application/Configuration/IncomingMessagesConfiguration.cs (hunk 1 above)
Comment (lasrinnil) [reply source/IncomingMessages.Application/Configuration/IncomingMessagesConfiguration.cs:44]: They could be, but for simplicity lets keep the current registration

File: source/IncomingMessages.Application/IncomingRequestAggregatedMeasuredParser.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        IArchivedMessagesClient archivedMessagesClient)
+    {
+        _requestAggregatedMeasureDataMarketMessageParser = requestAggregatedMeasureDataMarketMessageParser;
+        _incomingRequestAggregatedMeasuredDataSender = incomingRequestAggregatedMeasuredDataSender;
+        _aggregatedMeasureDataMarketMessageValidator = aggregatedMeasureDataMarketMessageValidator;
+        _responseFactory = responseFactory;
+        _archivedMessagesClient = archivedMessagesClient;
+    }
+
+    public async Task<ResponseMessage> ParseAsync(Stream message, DocumentFormat documentFormat, CancellationToken cancellationToken, DocumentFormat responseFormat = null!)
```
Comment (MadsDue) [inline source/IncomingMessages.Application/IncomingRequestAggregatedMeasuredParser.cs:54]: Make generic with a document type.

Comment (lasrinnil) [reply source/IncomingMessages.Application/IncomingRequestAggregatedMeasuredParser.cs]: done :)

**Review by MadsDue — DISMISSED** [review]

**Review by MadsDue — APPROVED** [review]

**Pull request merged**